pandas/plotting/_core.py

```diff
@@ -410,8 +410,13 @@ def get_label(i):
 
         if self.orientation == 'vertical' or self.orientation is None:
             if self._need_to_set_index:
-                xticklabels = [get_label(x) for x in ax.get_xticks()]
+                xticks = ax.get_xticks()
+                xticklabels = [get_label(x) for x in xticks]
                 ax.set_xticklabels(xticklabels)
+                from matplotlib.ticker import FixedLocator, FixedFormatter
```

Comment:
can you put imports higher up in this function to avoid need to import in separate locations